app/src/main/kotlin/com/wire/android/ui/home/conversations/edit/EditMessageMenuItems.kt

```diff
@@ -49,39 +50,40 @@ fun EditMessageMenuItems(
 ): List<@Composable () -> Unit> {
     val localContext = LocalContext.current
 
-    val onCopyItemClick = remember(message) {
-        {
-            hideEditMessageMenu {
-                onCopyClick(
-                    (message.messageContent as UIMessageContent.TextMessage).messageBody.message.asString(
-                        localContext.resources
-                    )
-                )
+    val isComposite = remember(message.header.messageId) {
+        message.messageContent is UIMessageContent.Composite
+    }
+
+    val onCopyItemClick: (() -> Unit)? = remember(message.header.messageId) {
+        (message.messageContent as? Copyable)?.textToCopy(localContext.resources)?.let {
+            {
+                hideEditMessageMenu { onCopyClick(it) }
             }
         }
     }
```

Comment:
The idea here is only Copyable messages will have the copy option displayed, and if we want to implement it to other messages we can do so by simply implementing the interface and we will be good to go
because before it will crash if it was called for anything that is not text message